ruby_assignment1.rb

```diff
@@ -0,0 +1,186 @@
+
+  if sum == kap
+    puts "#{kap} : is a Kaprekar number"
+  else
+    puts "#{kap} : is not Kaprekar Number"
+  end
+end
+check = kaprekar_num?(9)
+puts check
+puts
+
+# Q5) Find occurrence of RUBY from string 1.
+STRING_1 = 'RUBY parses a file by looking for <br/> one of the special tags that tells it to start interpreting the text
+as RUBY code. The parser then executes all of the code it finds until it runs into a RUBY closing <br/> tag.'
+puts STRING_1.scan('RUBY').length
+puts
+
+# Q6) Find the position where RUBY occurs in the string 1.
+puts STRING_1.index('RUBY')
+puts
+
+# Q7) Create array of words in string 1 & print them using a recursive function.
+puts STRING_1.split(' ')
+puts
+
+# Q8) Capitalise string 1
+puts STRING_1.capitalize
+puts
+
+# Q9) Combine string 1 & 2.
+STRING_2 = "RUBY does not require (or support) explicit type definition in variable declaration; a variable's type is
+determined by the context in which the variable is used."
+puts STRING_1 + STRING_2
+puts
+
+# Q10) Print current date.
+puts Time.now.strftime('%d/%m%Y %H:%M')
+puts
+
+# Q11) print 12th Jan 2012
+puts Time.new(2012, 1, 12)
+puts
+
+# Q12) add 7 days in current date
+puts (Time.now + (60 * 60 * 24 * 7)).strftime('%d/%m/%Y')
+puts
+
+# Q13) Cut the string 1 into 4 parts & print it.
+puts STRING_1.split(' ', 4)
+puts
+
+# Q14) Divide the string 1 by occurrences of '.'. Combine the array in reverse word sequence.
+puts STRING_1.split('.').reverse
+puts
+
+# Q15) Remove the HTML characters from string.
+puts STRING_1.gsub(%r{</?[^>]*>}, '')
+puts
+
+# Q16) Print the 'RUBY' word from string 1 by traversing it using string functions
+STRING_1.scan(/[\w']+/) do |word|
+  puts word if word.eql? 'RUBY'
+end
+puts
+
+# Q17) Find the length of string 1 & 2.
+puts STRING_1.length
+puts STRING_2.length
+puts
+
+# Q18) Compare two dates. (12-04-2010 & 12-05-2011). Calculate the days between these two dates.
+puts (Time.mktime(2011, 0o5, 12) - Time.mktime(2010, 0o4, 12)).to_i / (24 * 60 * 60)
+puts
+
+# Q19) Print date after 20 days from current date
+puts (Time.now + (60 * 60 * 24 * 20)).strftime('%d/%m/%Y')
+puts
+
+# Q20) Print date in array format.
+puts Time.now.to_a
+
+# Q21) Print prime numbers from the given array.
+require 'prime'
+def str_traverse(arr)
+  arr.each do |num|
+    if Prime.prime?(num)
+      puts "#{num} is a prime number"
+    else
+      puts "#{num} is not a prime number"
+    end
+  end
+end
+str_traverse([2, 6, 3, 8, 4, 9, 1])
+puts
+
+# 22) Write a program for fuel indication.
+# Ex: 0 -> Out of fuel, 1-10 -> Low, Please fill, 11-30 -> Good for now, 31-50 -> Almost Full, > 50 -> Full.
+input = 24
+case input
+when 0
+  puts 'Out of fuel'
+when (1..10)
+  puts 'Low, Please fill'
+when (11..30)
+  puts 'Good for now'
+when (31..50)
+  puts 'Almost Full'
+when 50
+  puts 'Full'
+end
+puts
+
+# Q23) Swap the given values(without the third variable)
+def swap_num(rust, boost)
+  rust, boost = boost, rust
```

Comment:
You do not even need to assign here if you follow this approach. Please write a logic